packages/lisk-p2p/src/peer.ts

```diff
@@ -278,48 +280,63 @@ export class Peer extends EventEmitter {
-		return socketClusterClient.create({
+		const outboundSocket = socketClusterClient.create({
 			hostname: this._ipAddress,
 			port: this._wsPort,
 			query: JSON.stringify(this._nodeInfo),
 			autoConnect: false,
 		});
+
+		this._bindHandlersToOutboundSocket(outboundSocket);
+
+		return outboundSocket;
 	}
 
-	private _handlePeerInfo(request: ProtocolRPCRequest): void {
-		// Update peerInfo with the latest values from the remote peer.
-		// TODO ASAP: Validate and/or sanitize the request.data as a PeerInfo object.
-		/* tslint:disable:next-line: no-object-literal-type-assertion */
-		this._peerInfo = {
-			...this._peerInfo,
-			...request.data,
-		} as PeerInfo;
-		this.emit(EVENT_UPDATED_PEER_INFO, this._peerInfo);
+	// All event handlers for the outbound socket should be bound in this method.
+	private _bindHandlersToOutboundSocket(outboundSocket: SCClientSocket): void {
+		outboundSocket.on('connect', () => {
+			this.emit(EVENT_CONNECT_OUTBOUND);
+		});
 	}
 
-	private _startListeningForRPCs(
-		socket: SCServerSocket | SCClientSocket,
+	// All event handlers for the outbound socket should be unbound in this method.
+	/* tslint:disable-next-line:prefer-function-over-method*/
+	private _unbindHandlersFromOutboundSocket(
+		outboundSocket: SCClientSocket,
 	): void {
-		// If an existing listener is bound, remove it.
-		socket.off(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
-		socket.on(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
+		outboundSocket.off();
 	}
 
-	private _stopListeningForRPCs(socket: SCServerSocket | SCClientSocket): void {
-		socket.off(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
+	// All event handlers for the inbound socket should be bound in this method.
+	private _bindHandlersToInboundSocket(inboundSocket: SCServerSocket): void {
+		inboundSocket.on(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
+		inboundSocket.on(REMOTE_EVENT_MESSAGE, this._handleMessage);
 	}
 
-	private _startListeningForMessages(
-		socket: SCServerSocket | SCClientSocket,
+	// All event handlers for the inbound socket should be unbound in this method.
+	private _unbindHandlersFromInboundSocket(
+		inboundSocket: SCServerSocket,
 	): void {
-		// If an existing listener is bound, remove it.
-		socket.off(REMOTE_EVENT_RPC_REQUEST, this._handleMessage);
-		socket.on(REMOTE_EVENT_RPC_REQUEST, this._handleMessage);
+		inboundSocket.off(REMOTE_EVENT_RPC_REQUEST, this._handleRPC);
+		inboundSocket.off(REMOTE_EVENT_MESSAGE, this._handleMessage);
 	}
 
-	private _stopListeningForMessages(
-		socket: SCServerSocket | SCClientSocket,
-	): void {
-		socket.off(REMOTE_EVENT_RPC_REQUEST, this._handleMessage);
+	private _handlePeerInfo(request: ProtocolRPCRequest): void {
+		// Update peerInfo with the latest values from the remote peer.
+		// TODO ASAP: Validate and/or sanitize the request.data as a PeerInfo object.
+		try {
+			const newPeerInfo = sanitizePeerInfo(request.data);
+			// Merge new info with existing info.
+			this._peerInfo = {
```

Comment:
I think it should only allow changing few values, in this case however even the `ipAddress`  and `port` could be updated. Maybe only height, version should be allowed to be updated?